control-plane-operator/controllers/hostedcontrolplane/cno/clusternetworkoperator.go

```diff
@@ -96,8 +96,13 @@ func NewParams(hcp *hyperv1.HostedControlPlane, version string, images map[strin
 	p.DeploymentConfig.SetControlPlaneIsolation(hcp)
 	p.DeploymentConfig.Replicas = 1
 	p.DeploymentConfig.SetDefaultSecurityContext = setDefaultSecurityContext
-	p.APIServerAddress = hcp.Status.ControlPlaneEndpoint.Host
-	p.APIServerPort = hcp.Status.ControlPlaneEndpoint.Port
+	if util.IsPrivateHCP(hcp) {
+		p.APIServerAddress = fmt.Sprintf("api.%s.hypershift.local", hcp.Name)
```

Comment:
just so I understand, does this resolve to the same we have now in `hcp.Status.ControlPlaneEndpoint.Host`?

Reply:
Yes, the only case that is change out of `Public`, PublicAndPrivate` and `Private` is `PublicAndPrivate` where it now uses the privatelink DNS name.